### DISCUSSION TITLE
Avatar not displayed

### DISCUSSION THREAD

**Discussion by maharjanPranish** [community/community#130956] [New to GitHub]
recently changed my primary email in github but when I log into git in my machine using my new email it doesn't show me my avatar in commits but when I log using the old email which is no longer associated with my github account, it recognizes me and shows the avatar. Please note that my email is not set to private.

In the picture, the commits with my avatar are done in git using my old email address after I have changed it but the one with Y is done with the new email address. I am switching email using the "git config user.email myEmail" command
![image](https://github.com/community/community/assets/133112545/d2ee39c5-cc18-4557-8cfd-a5c0cebb6288)

**Comment by dev-aditya-lab** [comment]
this is a common problem you can fix it to follow these steps 🙂
### Step 1: Verify Email Association in GitHub
1. **Log into GitHub:**
   - Go to [GitHub.com](https://github.com) and log in.
2. **Check Emails:**
   - Navigate to `Settings` -> `Emails`.
   - Ensure that your new email address is verified and associated with your GitHub account. If it’s not, add and verify it.

### Step 2: Update Git Configuration
1. **Set Your Global Email Configuration:**
   - Open your terminal.
   - Run the following command to set your global Git email configuration to your new email address:
     ```sh
     git config --global user.email "newEmail@example.com"
     ```

2. **Check the Current Email Configuration:**
   - Verify that your new email is set globally by running:
     ```sh
     git config --global user.email
     ```

### Step 3: Update Local Repository Configuration
1. **Set Email for a Specific Repository:**
   - Navigate to your repository directory.
   - Run the following command to set your email for this specific repository:
     ```sh
     git config user.email "newEmail@example.com"
     ```

2. **Check the Current Repository Email Configuration:**
   - Verify the email configuration for the repository by running:
     ```sh
     git config user.email
     ```

### Step 4: Amend the Latest Commit
1. **Amend the Last Commit with the New Email:**
   - If your latest commit was made with the old email, you can amend it to use the new email:
     ```sh
     git commit --amend --author="Your Name <newEmail@example.com>"
     ```
   - If you want to avoid changing the commit message, you can do:
     ```sh
     git commit --amend --no-edit --author="Your Name <newEmail@example.com>"
     ```

2. **Push the Amended Commit:**
   - Push the amended commit to the remote repository:
     ```sh
     git push --force
     ```

### Step 5: Update Past Commits (Optional)
1. **Rewriting Git History (Dangerous Operation):**
   - If you need to change the email for past commits, you can use a script to rewrite the history. This should be done with caution as it changes the commit history.
   - Create a script named `rewrite.sh`:
     ```sh
     #!/bin/sh
     git filter-branch --env-filter '
     OLD_EMAIL="oldEmail@example.com"
     CORRECT_NAME="Your Name"
     CORRECT_EMAIL="newEmail@example.com"
     if [ "$GIT_COMMITTER_EMAIL" = "$OLD_EMAIL" ]
     then
         export GIT_COMMITTER_NAME="$CORRECT_NAME"
         export GIT_COMMITTER_EMAIL="$CORRECT_EMAIL"
     fi
     if [ "$GIT_AUTHOR_EMAIL" = "$OLD_EMAIL" ]
     then
         export GIT_AUTHOR_NAME="$CORRECT_NAME"
         export GIT_AUTHOR_EMAIL="$CORRECT_EMAIL"
     fi
     ' --tag-name-filter cat -- --branches --tags
     ```
   - Make the script executable:
     ```sh
     chmod +x rewrite.sh
     ```
   - Run the script:
     ```sh
     ./rewrite.sh
     ```

2. **Force Push the Rewritten History:**
   - Force push the changes to the remote repository:
     ```sh
     git push --force --tags origin 'refs/heads/*'
     ```

### Step 6: Verify Changes on GitHub
1. **Check Your Commits:**
   - Go to your repository on GitHub.
   - Check the commit history to see if your avatar appears with the new email address.

### Step 7: Ensure Correct Email for Future Commits
1. **Consistent Email Configuration:**
   - Ensure that all your local repositories and your global Git configuration are set with your new email to avoid this issue in future commits.

**Comment by maharjanPranish** [comment]
i did as you mentioned but it only changed the commit with my old email that showed avatar into a commit with my new email that does not show avatar.

my old email was pranishmaharjan71@gmail.com and the commits with this email shows avatar.

my new email is pranishmaharjan071@gmail.com and the commits with this do not show my avatar.

pranishmaharjan071@gmail.com is the one that is connected to my github account with the avatar.

![image](https://github.com/community/community/assets/133112545/c67b14e1-22cf-49ba-9dd5-62bae8e0a5d2)

**Reply by dev-aditya-lab** [reply]
you verify your new email. if no chack you mail.
github send you a email verification link

**Reply by maharjanPranish** [reply]
it is already verified

![image](https://github.com/community/community/assets/133112545/08363deb-bc9e-44d0-9bfa-e34e1acd86c2)
